packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -226,6 +231,14 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           }
           modified[modifier] = true;
         }
+
+        if (
+          errorTemplate &&
+          disallowedModifiers &&
+          disallowedModifiers.includes(modifier)
+        ) {
+          this.raise(startPos, errorTemplate, modifier);
```

Comment:
nit: we can assume `errorTemplate` is non null if `disallowedModifiers` is provided. Overall this method not a public interface so we don't have to be defensive.